SmallBetterLogger/SmallBetterLogger.hpp

```diff
@@ -0,0 +1,179 @@
+		{
+			if (filepath.empty() 
+				|| filepath.size() < 1 
+				|| std::regex_match(filepath, std::regex(R"(^[ ]+$/g)"))) 
+				throw std::exception("No file path provided. Cannot create output stream.");
+
+			m_FileStream.open(filepath);
+			if (!m_FileStream.is_open()) throw std::exception("Cannot open file to output to.");
+		}
+
+		// Copy constructor
+		Logger(const Logger& other)
+			: m_StreamType(other.m_StreamType), m_AutoFlush(other.m_AutoFlush)
+		{ 
+			if (other.m_StreamType == STREAM_TYPE::FILE) throw std::exception("Cannot make a copy of a logger which outputs to a file stream.");
+		}
+
+		// Flush and close file stream if open
+		~Logger()
+		{
+			if (m_FileStream.is_open())
+			{
+				m_FileStream.flush();
+				m_FileStream.close();
+			}
+		}
+
+		// Public Methods
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist)
+		template<typename ...T>
+		inline void Write(const std::string& message, const T& ...t);
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist)
+		template<typename ...T>
+		inline void Write(const std::string&& message, const T&& ...t);
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist) and finishes with the newline character
+		template<typename ...T>
+		inline void WriteLine(const std::string& message, const T& ...t);
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist) and finishes with the newline character
+		template<typename ...T>
+		inline void WriteLine(const std::string&& message, const T&& ...t);
+
+	private:
+		// Private members
+		STREAM_TYPE m_StreamType;
+		std::fstream m_FileStream;
+		bool m_AutoFlush;
+
+		// Private methods
+
+		// Converts a T value to a string to be used in writing a log
+		template<typename T>
+		inline std::string stringConvert(const T& t);
+
+		// Replace all "{n}" placeholders with their respective values (n=0,...)
+		inline std::string replacePlaceholders(std::string message, std::vector<std::string>& items);
+
+		// Writes string to appropriate stream based on instance STREAM_TYPE (m_StreamType)
+		inline void writeToStream(const std::string&& str);
+	};
+
+	template<typename ...T>
+	inline void Logger::Write(const std::string& message, const T& ...t)
+	{
+		std::vector<std::string> printValues { stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues));
+	}
+
+	template<typename ...T>
+	inline void Logger::Write(const std::string&& message, const T&& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues));
+	}
+
+	template<typename ...T>
+	inline void Logger::WriteLine(const std::string& message, const T& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues) + "\n");
```

Comment:
Maybe use a macro for the newline character
```suggestion
		#ifdef _WIN32
    #define NEWLINE "\r\n"
#elif defined macintosh // OS 9
    #define NEWLINE "\r"
#else
    #define NEWLINE "\n" // Mac OS X uses \n
#endif
```